src/copy.ml

```diff
@@ -1037,28 +1031,18 @@ let transferFileUsingExternalCopyprog
     else
       Prefs.read copyprog
   in
-  let extraquotes = Prefs.read copyquoterem = `True
-                 || (  Prefs.read copyquoterem = `Default
-                    && Util.findsubstring "rsync" prog <> None) in
-  let addquotes root s =
-    match root with
-    | Common.Local, _ -> s
-    | Common.Remote _, _ -> if extraquotes then Uutil.quotes s else s in
+  let progTokens = Str.split (Str.regexp "[ \t]+") prog in
```

Comment:
Just a nit: Since this code is executed for every transfer then it may be beneficial to compute `Str.regexp "[ \t]+"` only once (make it a top-level value). Probably not worth it...

Since you don't use progTokens until below in the code, just move this line there.